include/vcpkg/sourceparagraph.h

```diff
@@ -218,10 +218,12 @@ namespace vcpkg
 
     void print_error_message(const LocalizedString& message);
 
-    std::string parse_spdx_license_expression(StringView sv, ParseMessages& messages);
-
-    // Exposed for testing
+    Optional<std::string> parse_spdx_license_expression(DiagnosticContext& context, StringView sv);
+    Optional<std::vector<Dependency>> parse_dependencies_list_context(DiagnosticContext& context,
+                                                                      const std::string& str,
+                                                                      StringView origin,
+                                                                      TextRowCol position = {1, 1});
     ExpectedL<std::vector<Dependency>> parse_dependencies_list(const std::string& str,
                                                                StringView origin,
-                                                               TextRowCol textrowcol = {});
+                                                               TextRowCol position = {1, 1});
```

Comment:
I think this should either not be defaulted OR should default to `{}` (no position information). Assuming that this is at the beginning of the file doesn't seem appropriate.